### PR TITLE
[AAE-10102] Storybook stories for Buttons Menu component

### PR DESCRIPTION
**Pull request by marcol13**
**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

[https://alfresco.atlassian.net/browse/AAE-10102](https://alfresco.atlassian.net/browse/AAE-10102)

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

In this PR I added three stories in Storybook for [Buttons menu](https://github.com/Alfresco/alfresco-ng2-components/blob/develop/docs/core/components/buttons-menu.component.md) component. I had some problems handling with `<ng-content></ng-content>` inside component. I have to use `template` property in Storybook which allow me to use content projection. Unfortunately `@ContentChildren` inside `.component.ts` don't work correctly with that so I hardcoded `isMenuEmpty` variable for every story and now everything is fine. 

I don't like this solution, so If someone has any better idea how to get around this problem please share with me. I can also make this PR a draft or come back to this problem when I will be more experienced with building stories.

Stories:
- Six Buttons
image

- One Button
image

- Zero Buttons
image

### CODE REVIEW

**Review by MichalFidor — COMMENTED** [review]
I'm not sure, what arrangements were made before you've started the work on buttons - stories are totally made by you? Or someone tells you what stories there should be? 

Anyway, I have 2 things:
1) Please, give each button other name than `settings` for each. When it comes to testing this, I'll need to targeting the buttons by adding the `nth` to locators, and I don't want to do that. :D  
2) I think that there are missing the stories for button disabled or hidden...?

**Comment by marcol13** [comment]
@MichalFidor I was assigned to task where I should test functionality and behavior of Buttons menu component in isolation making stories for 6 buttons, 1 button and 0 buttons. Whole task is described here: [[AAE-10102]](https://alfresco.atlassian.net/browse/AAE-10102).

About 1: That's a good point! I didn't think about it. I will change that! 😁
About 2: I didn't get task to make buttons disabled/hidden 😅 I can add proper stories for that if you need that functionality here 😃 But to be honest I don't know which cases it will cover 😅

If you have any other suggestions to stories/code you can share it with me, I can improve that stories file :P

**Comment by marcol13** [comment]
I realized that I can add `args` toggle button for `isMobile()` function so I did it in [c01a405](https://github.com/Alfresco/alfresco-ng2-components/pull/7731/commits/c01a40590fd168285e12327d14dd177527a91e7a)

File: lib/core/buttons-menu/buttons-menu.component.stories.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import { ButtonsMenuComponent } from './buttons-menu.component';
+import { ButtonsMenuModule } from './buttons-menu.module';
+import { MatIconModule } from '@angular/material/icon';
+
+export default {
+    component: ButtonsMenuComponent,
+    title: 'Core/Components/ButtonsMenu',
+    decorators: [
+        moduleMetadata({
+            imports: [CoreStoryModule, ButtonsMenuModule, MatIconModule]
```
Comment (tomgny) [inline lib/core/buttons-menu/buttons-menu.component.stories.ts:29]: please move MatIconModule to ButtonsMenuModule and remove MaterialModule. we should use only modules that we need
https://alfresco.atlassian.net/jira/software/c/projects/HXENG/boards/868?modal=detail&selectedIssue=HXENG-74

**Review by tomgny — APPROVED** [review]

**Comment by marcol13** [comment]
I have some weird issues related with styling story component. 


https://user-images.githubusercontent.com/56632321/182850150-9a47ebc6-6011-40aa-81b5-544f2bcc395d.mp4



I will wait until we upgrade Storybook version - maybe it will helps.
For now I convert this Pull Request to draft.

**Pull request merged**